Add release notes for `v0.2.1` (#1922)
VirtusLab/scala-cli: website/docs/release_notes.md

# Release notes

## [v0.2.1](https://github.com/VirtusLab/scala-cli/releases/tag/v0.2.1)

### Add a guide for migrating from the old `scala` runner to Scala CLI

As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
command. To make the transition smooth we added a [guide](docs/guides/old-runner-migration.md) highlighting
the differences between the two runners.

Added by [@Gedochao](https://github.com/Gedochao)  in [#1900](https://github.com/VirtusLab/scala-cli/pull/1900)

> ### Improve the `publish` and `publish setup` user experience

Reviewer: NIT
```suggestion
### Improve the `publish` and `publish setup` sub-commands' user experience
```

> We're currently focusing on improving the experimental `publish` feature of Scala CLI and making the UX of `publish setup` + `publish` better.

Reviewer: Can we give examples on what was improved?

Addressed by [@alexarchambault](https://github.com/alexarchambault) in [#1432](https://github.com/VirtusLab/scala-cli/pull/1432)
and by [@MaciejG604](https://github.com/MaciejG604) in [#1898](https://github.com/VirtusLab/scala-cli/pull/1898)

### Remove unsupported kebab-case style in using directives

All using directives names are now using camelCase, kebab-case is no longer available.

Added by [@lwronski](https://github.com/lwronski) in [#1878](https://github.com/VirtusLab/scala-cli/pull/1878)

### Add a reference for available config keys in help & docs

You can now view the available config keys using `config --help`:

> ```

Reviewer: ````suggestion
```bash
````

scala-cli config -h
# Usage: scala-cli config [options]
# Configure global settings for Scala CLI.
# 
# Available keys:
#   actions                                        Globally enables actionable diagnostics. Enabled by default.
#   interactive                                    Globally enables interactive mode (the '--interactive' flag).
#   power                                          Globally enables power mode (the '--power' launcher flag).
#   suppress-warning.directives-in-multiple-files  Globally suppresses warnings about directives declared in multiple source files.
#   suppress-warning.outdated-dependencies-files   Globally suppresses warnings about outdated dependencies.
# 
# You are currently viewing the basic help for the config sub-command. You can view the full help by running: 
#    scala-cli config --help-full
# For detailed documentation refer to our website: https://scala-cli.virtuslab.org/docs/commands/misc/config
# 
# Config options:
#   --unset, --remove  Remove an entry from config
```

> Also, `config --full-help` will show the list of all keys, including the ones used in commands that require `--power`.

Reviewer: ```suggestion
Also, `config --full-help` will show the list of all keys.
```
`--power` is actually irrelevant here and will be addressed separately in a future PR.

Added by [@Gedochao](https://github.com/Gedochao)  in [#1910](https://github.com/VirtusLab/scala-cli/pull/1910)

### Pass user arguments to JS runner

It's now possible to pass user arguments to a JS application:

> ```scala ignore
> import scala.scalajs.js
> import scala.scalajs.js.Dynamic.global
> 
> val process = global.require("process")
> val argv = Option(process.argv)
>   .filterNot(js.isUndefined)
>   .map(_.asInstanceOf[js.Array[String]].drop(2).toSeq)
>   .getOrElse(Nil)
> val console = global.console
> console.log(argv.mkString(" "))
> ```
> 
> 
> ```bash ignore
> scala-cli . --js -- Hello World
> ```

Reviewer: Why ignore?
Won't this work:
````suggestion
```scala title=ScalaJsArgs.scala
import scala.scalajs.js
import scala.scalajs.js.Dynamic.global

val process = global.require("process")
val argv = Option(process.argv)
  .filterNot(js.isUndefined)
  .map(_.asInstanceOf[js.Array[String]].drop(2).toSeq)
  .getOrElse(Nil)
val console = global.console
console.log(argv.mkString(" "))
```

```bash
scala-cli ScalaJsArgs.scala --js -- Hello World
```
````
?

```text
Hello World
```

Added by [@alexarchambault](https://github.com/alexarchambault) in [#1826](https://github.com/VirtusLab/scala-cli/pull/1826)

### Other changes

> * Ensure no console-syntax in reference docs and no `md` fenced blocks in `--help` by [@Gedochao](https://github.com/Gedochao)  in [#1874](https://github.com/VirtusLab/scala-cli/pull/1874)

Reviewer: I'd treat this as a documentation change

* Tweak error messages for running scripts without file extensions by [@Gedochao](https://github.com/Gedochao)  in [#1886](https://github.com/VirtusLab/scala-cli/pull/1886)
* Exit with Bloop command return code if it's non-zero by [@alexarchambault](https://github.com/alexarchambault) in [#1837](https://github.com/VirtusLab/scala-cli/pull/1837)
* bloop-rifle: increase timeout values by [@Flowdalic](https://github.com/Flowdalic) in [#1865](https://github.com/VirtusLab/scala-cli/pull/1865)

> * Suggest users to clean working directory when Nailgun server failed by [@lwronski](https://github.com/lwronski) in [#1916](https://github.com/VirtusLab/scala-cli/pull/1916)

Reviewer: I'd treat all these as fixes

# Release notes

## [v0.2.1](https://github.com/VirtusLab/scala-cli/releases/tag/v0.2.1)

> ### Add a guide for migrating from old `scala` runner to Scala CLI

Reviewer: ```suggestion
### Add a guide for migrating from the old `scala` runner to Scala CLI
```

> As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
> command. To make the transition smooth [@Gedochao](https://github.com/Gedochao)  prepared a [guide](./guides/old-runner-migration.md) highlighting
> the differences between the two runners.

Reviewer: ```suggestion
As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
command. To make the transition smooth, we added a [guide](./guides/old-runner-migration.md) highlighting the differences between the two runners.
```
Make sure to use an absolute link to the guide on the actual release notes, as a relative link within the repo won't work there.
